target/linux/rockchip/image/Makefile

```diff
@@ -0,0 +1,51 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+FAT32_BLOCK_SIZE=1024
```

Comment:
Does the bootflow require the first partition to be FAT formatted? If not, the preference would be using `scripts/gen_image_generic.sh`, instead of introducing another device specific script. One exception would probably be EFI boot, if someone would like to add support for such images.

Reply:
No, we don't need FAT, we could also use ext4 or something else.
But what we do need is the SoC-specific script that dd's the bootloader to magic locations on the MMC image. So I fear we can't get rid of the script yet, sorry.

Reply:
An example here: https://github.com/openwrt/openwrt/blob/0495324b9bd0b6a976ea731a1b4b77b26df32ee6/target/linux/tegra/image/Makefile#L33
You could add it as separate command `uboot-install`.

Reply:
Switched to EXT4.